packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,556 @@
+    pulp_host: https://pulp.example.com
+    repo_type: rpm
+    feed_url: http://mirror.centos.org/centos/6/updates/x86_64/
+    relative_url: centos/6/updates
+    url_username: admin
+    url_password: admin
+    force_basic_auth: yes
+    state: present
+
+- name: Remove a repo from the pulp server
+  pulp_repo:
+    name: my_old_repo
+    repo_type: rpm
+    state: absent
+'''
+
+RETURN = '''
+repo:
+  description: Name of the repo that the action was performed on.
+  returned: success
+  type: string
+  sample: repo_name
+'''
+
+
+
+
+class pulp_server(object):
+    """
+    Class to interact with a Pulp server
+    """
+
+    def __init__(self, module, pulp_host, repo_type):
+        self.module = module
+        self.host = pulp_host
+        self.repo_type = repo_type
+        self.repo_cache = dict()
+
+    def check_repo_exists(self, repo_id):
+        try:
+            repo_config = self.get_repo_config_by_id(repo_id)
+
+        except IndexError:
+            return False
+
+        else:
+            return True
+
+    def compare_repo_distributor_config(self, repo_id, **kwargs):
+        repo_config = self.get_repo_config_by_id(repo_id)
+
+        for distributor in repo_config['distributors']:
+            for key, value in kwargs.iteritems():
+                if not distributor['config'][key] == value:
+                    return False
+
+        return True
+
+    def compare_repo_importer_config(self, repo_id, **kwargs):
+        repo_config = self.get_repo_config_by_id(repo_id)
+
+        for importer in repo_config['importers']:
+            for key, value in kwargs.iteritems():
+                if value is not None:
+                    if key not in importer['config'].keys():
+                        return False
+
+                    if not importer['config'][key] == value:
+                        return False
+
+        return True
+
+    def create_repo(
+        self,
+        repo_id,
+        relative_url,
+        feed_url=None,
+        serve_http=False,
+        serve_https=True,
+        proxy_host=None,
+        proxy_port=None,
+        ssl_ca_cert=None,
+        ssl_client_cert=None,
+        ssl_client_key=None
+    ):
+        url = "%s/pulp/api/v2/repositories/" % self.host
+        data = dict()
+        data['id'] = repo_id
+        data['distributors'] = []
+
+        if self.repo_type == 'rpm':
+            yum_distributor = dict()
+            yum_distributor['distributor_id'] = "yum_distributor"
+            yum_distributor['distributor_type_id'] = "yum_distributor"
+            yum_distributor['auto_publish'] = True
+            yum_distributor['distributor_config'] = dict()
+            yum_distributor['distributor_config']['http'] = serve_http
+            yum_distributor['distributor_config']['https'] = serve_https
+            yum_distributor['distributor_config']['relative_url'] = relative_url
+            data['distributors'].append(yum_distributor)
+
+            export_distributor = dict()
```

Comment:
Please could you make the list of distributors optional? See https://github.com/ansible/ansible-modules-extras/pull/1961#issuecomment-228165780.